framework/contracts/account/src/reply.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{
+    contract::{AccountResponse, AccountResult},
+    modules::INSTALL_MODULES_CONTEXT,
+};
+use abstract_std::objects::{
+    module::{assert_module_data_validity, Module},
+    module_reference::ModuleReference,
+};
+use cosmwasm_std::{DepsMut, Reply, Response, StdError};
+
+/// Add the message's data to the response
+pub fn forward_response_data(result: Reply) -> AccountResult {
+    // get the result from the reply
+    let res = result.result.into_result().map_err(StdError::generic_err)?;
+
+    // log and add data if needed
+    let resp = if let Some(data) = res.data {
+        AccountResponse::new(
+            "forward_response_data_reply",
+            vec![("response_data", "true")],
+        )
+        .set_data(data)
+    } else {
+        AccountResponse::new(
+            "forward_response_data_reply",
+            vec![("response_data", "false")],
+        )
+    };
+
+    Ok(resp)
+}
```

Comment:
@CyberHoward do we even need this method in cosmwasm 2? Won't it be possible to get data from the new field of SubMsgResponse: https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.SubMsgResponse.html#structfield.msg_responses? Let's mark this method as deprecated for reminder as well if so

Reply:
I think it's the same behavior here, because it's still a nested execution

Reply:
Asked for clarification here : https://github.com/CosmWasm/cosmwasm/issues/2253

Reply:
> @CyberHoward do we even need this method in cosmwasm 2? Won't it be possible to get data from the new field of SubMsgResponse: https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.SubMsgResponse.html#structfield.msg_responses? Let's mark this method as deprecated for reminder as well if so

Yeah I removed the `allow(deprecated)`

But we should do this in a different PR imo.